src/components/SeatLayout/SeatLayout.tsx

```diff
@@ -0,0 +1,133 @@
+import { deckLayoutProducer, layoutConfig } from './seatConfig';
+import SeatLayoutWrapper from './SeatLayout.styled';
+import driverSeat from '../../assets/tabler_steering-wheel.svg';
+import { ISeat, ISeatStatus } from '../../api/types/trip';
+import { Tooltip } from '@mui/material';
+import { useEffect, useState } from 'react';
+
+// TODO: To be removed.
+// When calling the SeatLayout component, this state variable is to be declared
+// in the parent component and then passed as prop to the SeatLayout component.
+//
+// This state contains the seats selected by a user
+// const [selectedSeats, setSelectedSeats] = useState<number[]>([]);
+
+// This function add/removes seats to/from the selectedSeat[]
+// const updateSelectedSeats = (newSeat: number) => {
+//     if (selectedSeats.includes(newSeat)) {
+//         setSelectedSeats((prev) =>
+//             prev.filter((selectedSeat) => selectedSeat != newSeat)
+//         );
+//     } else setSelectedSeats((prev) => [...prev, newSeat]);
+// };
+//
+// This is how the seatLayout component is to be called when required.
+// <SeatLayout
+//     layoutName={layoutNames.volvo25}
+//     seats={seats}
+//     selectedSeats={selectedSeats}
+//     updateSelectedSeats={updateSelectedSeats}
+// />
+
+const SeatLayout = ({
+    layoutName,
+    seats,
+    selectedSeats,
+    updateSelectedSeats,
+}: {
+    layoutName: string;
+    seats: ISeat[];
+    selectedSeats: number[];
+    updateSelectedSeats: (seat: number) => void;
+}) => {
+    const [deck, setDeck] = useState<{
+        lowerDeck: number[][];
+    }>({ lowerDeck: [] });
+
+    let seatIndex = 0;
+
+    useEffect(() => {
+        setDeck((prev) => ({
+            ...prev,
+            lowerDeck: deckLayoutProducer(layoutConfig[layoutName].lowerDeck),
+        }));
+    }, [layoutName]);
+
+    return (
+        <SeatLayoutWrapper>
+            <img src={driverSeat} alt="driver seat" className="driver" />
```

Comment:
i think src, alt and class names based on steering wheel will be more apt since the image is a steering wheel icon.